### DISCUSSION TITLE
someting wrong in running ICAFixProcessingBatch.sh

### DISCUSSION THREAD

**Discussion by betsy1226** [Washington-University/HCPpipelines#293] [Q&A]
I get an error when I do ./ICAFixProcessingBatch.sh
It said

hcp_fix: Done running FIX
Could not find a supported file with prefix "rfMRI_REST1_RL_hp2000.ica/filtered_func_data_clean"
hcp_fix: ERROR: '/usr/local/fsl/bin/immv' command failed with return code: 1
===> ERROR: Command returned with nonzero exit code
---------------------------------------------------
         script: hcp_fix
stopped at line: 462
           call: $FSLDIR/bin/immv ${fmrihp}.ica/filtered_func_data_clean ${fmrihp}_clean
  expanded call: /usr/local/fsl/bin/immv rfMRI_REST1_RL_hp2000.ica/filtered_func_data_clean rfMRI_REST1_RL_hp2000_clean
      exit code: 1

I used Matlab2020b and chose FSL_FIX_MATLAB_MODE=1 since ubuntu20.04 supports 2020b or higher version.
I am confused that Fix only could be used in Matlab 2017? or there were other possible reasons?

**Comment by glasserm** [comment]
I would have a look at the FIX documentation and some of the internal FIX logs.

**Reply by betsy1226** [reply]
In the file rfMRI_REST1_RL_hp2000.ica, there was a txt named "logMatlab.txt"
It said


rfMRI_REST1_RL_hp2000.ica/
        67         1
Error Time: 05/22/2024 22:23:19
'aryule' undefined near line 11 column 22
 
rfMRI_REST1_RL_hp2000.ica/
        67         1
Error Time: 05/22/2024 22:28:25
'aryule' undefined near line 11 column 22

**Comment by glasserm** [comment]
Looks like you are missing the Matlab signal processing toolbox.  You could run compiled matlab instead.

**Reply by betsy1226** [reply]
I had tried matlab runtime 2017, but there was something wrong in lib in Ubuntu 20.04.

**Comment by mharms** [comment]
That sounds like your installed version of Matlab is missing a Toolbox that ‘fix’ expects to exist.  Possibly the Signal Processing Toolbox?

--
Michael Harms, Ph.D.
-----------------------------------------------------------
Professor of Psychiatry
Washington University School of Medicine
Department of Psychiatry, Box 8134
660 South Euclid Ave.                        Tel: 314-747-6173
St. Louis, MO  63110                          Email: ***@***.***

From: betsy1226 ***@***.***>
Reply-To: Washington-University/HCPpipelines ***@***.***>
Date: Wednesday, May 22, 2024 at 10:15 AM
To: Washington-University/HCPpipelines ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [Washington-University/HCPpipelines] someting wrong in running ICAFixProcessingBatch.sh (Discussion #293)

* External Email - Caution *

In the file rfMRI_REST1_RL_hp2000.ica, there was a txt named "logMatlab.txt"
It said

rfMRI_REST1_RL_hp2000.ica/
67 1
Error Time: 05/22/2024 22:23:19
'aryule' undefined near line 11 column 22

rfMRI_REST1_RL_hp2000.ica/
67 1
Error Time: 05/22/2024 22:28:25
'aryule' undefined near line 11 column 22

—
Reply to this email directly, view it on GitHub<https://github.com/Washington-University/HCPpipelines/discussions/293#discussioncomment-9523492>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AA5TABR2UR7BI5MIW76NKTLZDSZATAVCNFSM6AAAAABID3BMMOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKMRTGQ4TE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________
The materials in this message are private and may contain Protected Healthcare Information or other information of a sensitive nature. If you are not the intended recipient, be advised that any unauthorized use, disclosure, copying or the taking of any action in reliance on the contents of this information is strictly prohibited. If you have received this email in error, please immediately notify the sender via telephone or return mail.

**Reply by betsy1226** [reply]
Thanks for your suggestion. I tested the Signal Processing Toolbox and function 'aryule'. Signal Processing Toolbox seemed to work well. I also tried to work on Matlab2022b, also failed.
